curriculum/challenges/english/25-front-end-development/review-react-basics/67487e141bb6a7140a352e12.md

```diff
@@ -0,0 +1,253 @@
+
+## React Components
+
+- Components are the building blocks of React applications that allow developers to break down complex user interfaces into smaller, manageable pieces.
+- The UI is described using JSX, a syntax extension that allows developers to write HTML-like code within JavaScript.
+- Components are basically JS functions or classes that return a piece of UI. 
+Here is an example of a simple React component that renders a greeting message:
+
+```jsx
+function Greeting() {
+    const name = 'Anna';
+    return <h1>Welcome, {name}!</h1>;
+}
+```
+
+To use the component, you can simply call:
+    
+```jsx
+    <Greeting />
+```
+
+## Meta frameworks
+
+- Meta frameworks, also known as full-stack frameworks, are tools that are build upon existing libraries or frameworks to provide additional features like server-side rendering (SSR), static site generation (SSG), routing, code splitting, and optimized build processes out of the box.
+- In the React ecosystem, Next.js is an example of a meta framework.
+
+Here is an example of a simple Next.js page that renders the same greeting message as the previous example:
+
+```jsx
+function Greeting() {
+    const name = 'Anna';
+    return <h1>Welcome, {name}!</h1>;
+}
+
+export default Greeting
+```
+
+- Meta frameworks can perform server-side rendering. This means that the initial HTML is rendered on the server, which can lead to faster page loads and improved SEO.
+
+## CLI tools for project setup
+
+- Project setup tools and CLIs provide a quick & easy way to start new projects, allowing developers to focus on writing code rather than dealing with configuration. 
+- Vite, a popular project setup tool, can be used with React, Vue, Svelte, and even vanilla JavaScript projects.
+- To create a new project with Vite, you can use the following command in your terminal:
+    
+```bash
+npm init vite@latest my-react-app -- --template react
+```
+
+This command creates a new React project named `my-react-app` using Vite's React template. 
+
+- Vite also provides a dev server to see the changes live on the browser.
+- Create React App (CRA), another widely used tool, is an officially supported way to create single-page React applications. To create a new project with CRA, you would use:
+
+```bash
+npx create-react-app my-react-app
+```
+
+- For Vue.js projects, the Vue CLI is a popular choice. You can start a new Vue project by running:
+
+```bash
+npm init vue@latest
+```
+
+This command will prompt you with questions about your project setup, allowing you to choose features like TypeScript support, routing, and state management.
+
+- Angular projects can be set up using the Angular CLI. To create a new Angular project, you can run:
+
+```bash
+ng new my-angular-app
+```
+
+- The Angular CLI will ask you some questions to customize your project setup, such as if you want to include routing and which CSS preprocessor you would like to use.
+
+## Dependency management in React
```

Comment:
This section needs to be removed because that lecture script was moved to another module

Reply:
removed